nipype/utils/filemanip.py

```diff
@@ -828,7 +827,7 @@ def dist_is_editable(dist):
     return False
 
 
-def makedirs(path, exist_ok=False):
+def makedirs(path, mode=0o777, exist_ok=False):
```

Comment:
Just to be safe, I'd add it to the end of the signature to avoid any wrong assignments. I can't wait til we move to py3 only to support keyword only args

Reply:
This keeps the protocol consistent with `os.makedirs` in all versions:

```Shell
$ pydoc2.7 os.makedirs | cat
Help on function makedirs in os:

os.makedirs = makedirs(name, mode=511)
    makedirs(path [, mode=0777])

$ pydoc os.makedirs | cat 
Help on function makedirs in os:

os.makedirs = makedirs(name, mode=511, exist_ok=False)
    makedirs(name [, mode=0o777][, exist_ok=False])
```

We only use this function with the keyword:

```
$ grep -rI makedirs\( nipype | grep -v os\.makedirs
nipype/pipeline/plugins/tools.py:    makedirs(crashdir, exist_ok=True)
nipype/pipeline/engine/utils.py:    makedirs(base_dir, exist_ok=True)
nipype/pipeline/engine/workflows.py:        base_dir = makedirs(base_dir, exist_ok=True)
nipype/pipeline/engine/workflows.py:        makedirs(report_dir, exist_ok=True)
nipype/pipeline/engine/nodes.py:        makedirs(outdir, exist_ok=True)
nipype/pipeline/engine/nodes.py:            makedirs(outdir, exist_ok=True)
nipype/utils/filemanip.py:            makedirs(str(self), mode=mode, exist_ok=exist_ok)
nipype/utils/filemanip.py:def makedirs(path, mode=0o777, exist_ok=False):
nipype/utils/filemanip.py:    makedirs(path)
```

Reply:
works for me - looks good